internal/bgp/bgp.go

```diff
@@ -665,3 +679,33 @@ func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
 
 	return t
 }
+
+// localAddressExists returns true if the provided address exists locally on
+// any of the host's network interfaces. The function doesn't return errors and
+// instead returns false if an error has occurred while looking for the
+// address. This function is safe to call if addr is nil or malformed.
+func localAddressExists(addr string) bool {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return false
+	}
+
+	for _, i := range interfaces {
+		addresses, err := i.Addrs()
+		if err != nil {
+			return false
+		}
+
```

Comment:
We are talking about the **source** IP here, which doesn't necessarily determine the egress interface. Think about the following situation:

- A machine has 2 NICs: NIC 1 has `10.0.0.1/24` configured and NIC 2 has `172.16.0.1/24` configured.
- NIC 2 is down.
- The default route points at `10.0.0.254` via NIC 1.

In this case, at least on my testing Linux seems to allow constructing a packet with a source of `172.16.0.1` and sending it on NIC 1. I'm not sure why a user would want to do so, but since it works I see no special reason actively block this option.

Reply:
This function is now removed.